### PR TITLE
[Hub Generated] Review request for Microsoft.CustomerLockbox to add version preview/2018-02-28-preview

### PR DESCRIPTION
**Pull request by vilaksh**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/vilaksh/azure-rest-api-specs/dev-customerlockbox-Microsoft.CustomerLockbox-2018-02-28-preview...master/).

There are some mismatches between the API contract that is currently exposed by the service and the swagger spec documented. This PR is to fix the documentation to match the contract exposed by the service.

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
 -- N/A: The contract was incorrect and flagged in S360, hence fixing the same. The service contract hasn't changed recently, the documentation was incorrect so far.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
-- There are no changes to the service, fixing incorrect documentation.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by vilaksh** [comment]
There are some mismatches between the API contract that is currently exposed by the service and the swagger spec documented. This PR is to fix the documentation to match the contract exposed by the service.

**Comment by PhoenixHe-msft** [comment]
@vilaksh You need to go through breaking change review board

**Comment by vilaksh** [comment]
> @vilaksh You need to go through breaking change review board

Thank you @PhoenixHe-msft for taking a look. The contract which we had originally published was incorrect. There are no changes that are happening at this time in the service, we are only updating the contract to match what is currently running. Can you please let us know if we still need to go through breaking changes review for this scenario.

**Comment by PhoenixHe-msft** [comment]
@vilaksh Yes it's still required task. And please fix ModelValidation issue.

**Comment by vilaksh** [comment]
> @vilaksh Yes it's still required task. And please fix ModelValidation issue.

Sounds good, I've started the process and will update once I hear back. I've asked for an exception for the model validation as well. Similar to other changes, this is the contract used by the service at this time (been this way for about a year, but we had the swagger contract published incorrectly). The PR is to fix the mismatch.

**Comment by vilaksh** [comment]
@jianyexi can you please take a look at this PR. We have gotten approval for breaking changes.

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE can you please take a look at this PR. We have gotten approval for breaking changes.

Could you fix the model validation issues ?

**Comment by vilaksh** [comment]
> > @jianyexi Jianye Xi FTE can you please take a look at this PR. We have gotten approval for breaking changes.
> 
> Could you fix the model validation issues ?

@jianyexi , The model validation errors are flagging new properties that were added. These are currently returned by the service and the spec is incorrect. We reviewed these as part of breaking changes.

**Comment by vilaksh** [comment]
> Jianye Xi

Does the explanation for model validation sound reasonable? We reviewed these changes in the breaking changes scrum already.

File: specification/customerlockbox/resource-manager/Microsoft.CustomerLockbox/preview/2018-02-28-preview/examples/Requests_List_FilterByStatus.json [hunk 1]
```diff
@@ -16,17 +16,15 @@
...
-              "supportCaseUrl": "https://portal.azure.com/#resource/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/microsoft.support/supporttickets/123",
+              "status": "Expired",
+              "createdDateTime": "2020-06-17T23:15:42.031334Z",
+              "expirationDateTime": "2020-06-21T23:15:42.031334Z",
+              "duration": "8",
+              "resourceIds": "SUBSCRIPTIONS/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/RESOURCEGROUPS/DEMORG/PROVIDERS/MICROSOFT.COMPUTE/VIRTUALMACHINES/DEMO",
+              "resourceType": "VMSS",
+              "supportRequest": "12345",
+              "supportCaseUrl": "https://portal.azure.com/#resource/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/microsoft.support/supporttickets/12345",
+              "workitemsource": "Other",
```
Comment (jianyexi) [inline specification/customerlockbox/resource-manager/Microsoft.CustomerLockbox/preview/2018-02-28-preview/examples/Requests_List_FilterByStatus.json:27]:
```suggestion
              "workitemSource": "Other",
```

**Comment by jianyexi** [comment]
> > Jianye Xi
> 
> Does the explanation for model validation sound reasonable? We reviewed these changes in the breaking changes scrum already.

I think it's caused by wrong casing

**Comment by vilaksh** [comment]
> > > Jianye Xi
> > 
> > 
> > Does the explanation for model validation sound reasonable? We reviewed these changes in the breaking changes scrum already.
> 
> I think it's caused by wrong casing

Ah, thanks for pointing this out. I'm so sorry, my bad. I missed this - was thinking the error was referring to the fact that a new property was added. Updated the PR with a new iteration fixing this issue.

**Comment by vilaksh** [comment]
@jianyexi , can you please take a look at this PR. I have resolved the model validation issues.

**Comment by jianyexi** [comment]
@msyyc @qiaozha @ArcturusZhang  could you help with the SDK generation failure ?

**Comment by vilaksh** [comment]
@jianyexi , all the checks have been resolved now. Can you please take a look.

I've updated the PR to remove languages that are not used by our customers. Given there is no usage it won't be worth looking into these issues further and resolving the mismatches between spec and actual service contract is higher priority.

**Comment by vilaksh** [comment]
@jianyexi , can you please take a look - is there any more details required here. All checks have passed and we went through breaking changes process as well.

**Review by jianyexi — APPROVED** [review]

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE , can you please take a look - is there any more details required here. All checks have passed and we went through breaking changes process as well.

Ready for merging ?

**Comment by vilaksh** [comment]
> > @jianyexi Jianye Xi FTE , can you please take a look - is there any more details required here. All checks have passed and we went through breaking changes process as well.
> 
> Ready for merging ?

Yes, would you be authorized to merge these changes? Looks like I'm not authorized.

**Pull request merged**